pounders/py/tests/TestCheckInputss.py

```diff
@@ -24,94 +24,93 @@ def setUp(self):
         self.L = np.zeros(self.n)
         self.U = np.ones(self.n)
 
+    def __testCommonFinalConditions(self, out, flag):
+        if flag == "success":
+            self.assertEqual(out[0], 1, "Should not have failed")
+        elif flag == "fail":
+            self.assertEqual(out[0], -1, "Should have failed")
+        elif flag == "warn":
+            self.assertEqual(out[0], 0, "Should have warned, but not failed")
+
+        self.assertEqual(len(out), 7, "Should always have 7 outputs from checkinputss")
+
+    def test_checkinputts0(self):
+        out = checkinputss(self.fun, self.X0, self.n, self.npmax, self.nfmax, self.gtol, self.delta, self.nfs, self.m, self.F0, self.xkin, self.L, self.U)
+        self.__testCommonFinalConditions(out, "success")
+
     def test_checkinputts1(self):
         fun_to_fail = []
         out = checkinputss(fun_to_fail, self.X0, self.n, self.npmax, self.nfmax, self.gtol, self.delta, self.nfs, self.m, self.F0, self.xkin, self.L, self.U)
-        self.assertEqual(out[0], -1, "Should have failed")
-        self.assertEqual(len(out), 7, "Should always have 7 outputs from checkinputss")
+        self.__testCommonFinalConditions(out, "fail")
```

Comment:
I thought that this would be overkill for your test, but it cleaned up the tests nicely and boosted readability.

Reply:
I agree it both cleaned the test and improved readability